runtime/LibcWrappers.cpp

```diff
@@ -385,4 +385,298 @@ int SYM(memcmp)(const void *a, const void *b, size_t n) {
+    auto *tailExpr_5 = _sym_build_integer(53, 8);  // '5'
+    auto *tailExpr_6 = _sym_build_integer(54, 8);  // '6'
+    auto *tailExpr_7 = _sym_build_integer(55, 8);  // '7'
+    auto *tailExpr_8 = _sym_build_integer(56, 8);  // '8'
+    auto *tailExpr_9 = _sym_build_integer(57, 8);  // '9'
+
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_0),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_1),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_2),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_3),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_4),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_5),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_6),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_7),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_8),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+    _sym_push_path_constraint(
+            _sym_build_not_equal(*shadowIt, tailExpr_9),
+            /*taken*/ 1, reinterpret_cast<uintptr_t>(SYM(strchr)));
+
+    return result;
+}
+
+char *SYM(strcpy)(char *dest, const char *src) {
+    tryAlternative(dest, _sym_get_parameter_expression(0), SYM(strcpy));
+    tryAlternative(src, _sym_get_parameter_expression(1), SYM(strcpy));
+
+    auto *result = strcpy(dest, src);
+    _sym_set_return_expression(nullptr);
+
+    size_t cpyLen = strlen(src);
+    if (isConcrete(src, cpyLen) && isConcrete(dest, cpyLen))
+        return result;
+
+    auto srcShadow = ReadOnlyShadow(src, cpyLen);
+    auto destShadow = ReadWriteShadow(dest, cpyLen);
+
+    std::copy(srcShadow.begin(), srcShadow.end(), destShadow.begin());
+
+    return result;
+}
+
+char *SYM(strncpy)(char *dest, const char *src, size_t n) {
+    tryAlternative(dest, _sym_get_parameter_expression(0), SYM(strncpy));
+    tryAlternative(src, _sym_get_parameter_expression(1), SYM(strncpy));
+    tryAlternative(n, _sym_get_parameter_expression(2), SYM(strncpy));
+
+    auto *result = strncpy(dest, src, n);
+    _sym_set_return_expression(nullptr);
+
+    size_t srcLen = strnlen(src, n);
+    size_t copied = std::min(n, srcLen);
```

Comment:
This is really minor, but isn't `srcLen` always less than or equal to `n` as per the definition of `strnlen`?